### PR TITLE
Add functionality for dynamic pool reconfiguration

### PR DESCRIPTION
**Pull request by pbrownlow7**
Adds three files (reconfigure_setup, reconfigure, reaffinitize) to intel/ that create a functionality with CMK to reconfigure the pool setup without having to tear down CMK and build it back up. Reassigns running processes to the new cores in the reconfigured setup so the user does not have to stop all processes before running reconfigure.

Will add more details about what is actually happening when reconfiguration is deployed.

Also fixes the linting issue that was stopping the repo from passing the tests (problem in util.py)

### CODE REVIEW

**Comment by pbrownlow7** [comment]
Hijacking this PR to also add the changes to CMK's configuration bookkeeping. Instead of using the directory system to keep track of all the cores on a node, use a configmap (named after the node it's associated with) that the pods can access using the service account cmk-serviceaccount provided with the repo. The configmap for each node is created by the init container and a pod wishing to use a configmap will work out which node it is on to get the correct one.

To account for simultaneous calls to update a configmap, after a pod gets the configmap, it deletes it before changing and recreating it under the same name. A pod will wait a certain amount of time while trying to retrieve the configmap before giving up in case another pod is using it.

**Review by przemeklal — COMMENTED** [review]
MTLS changes look okay to me, the only things that may be missing:
- currently we don't have any code that configures kube-apiserver to send its own client cert to our webhook, so maybe either an automation of that setup or at least a small piece of documentation how to force Kubernetes API server to send its own client certificate to the webhook service (a link to the official doc would do I think)
- there's no way to disable MTLS like an --insecure flag or something if possible - of course MTLS will be enabled by default

I didn't try to run it yet, by I assume that after cluster-init the webhook would be rejecting all AdmissionReview requests sent by the apiserver due to missing client certificate.

**Comment by pbrownlow7** [comment]
Yeah I think the best way to go about it would be just the update the documentation and advise the user on how to set up their kube apiserver to get it to send the correct certificates to the webhook., I'll update the docs accordingly.

I'll go about adding the '--insecure' functionality as well.

**Comment by pbrownlow7** [comment]
With the --insecure flag the user can decide whether they want mTLS to be activated. If mTLS is activated and the kube-apiserver is not set up correctly to pass certificates to any webhooks, any pods created will be stuck in pending. With the --insecure flag set to 'True', pods will be created as normal. Just need to update the documentation for setting up the kube-apiserver correctly

File: intel/util.py [hunk 1]
```diff
@@ -74,6 +74,8 @@ def generate_cert(service, namespace, private_key):
         x509.DNSName("{0}.{1}".format(service, namespace)),
         x509.DNSName("{0}.{1}.svc".format(service, namespace)),
         x509.DNSName("localhost"),
+        x509.DNSName("silpixa00399862.ir.intel.com"),
```
Comment (przemeklal) [inline intel/util.py:77]: :eyes:

**Review by przemeklal — COMMENTED** [review]
I don't see any obvious problems with the newly added changes, LGTM except for the hardcoded hostname. In the future we might consider switching from configmap to CRD objects, but for now it should be okay.

**Comment by pbrownlow7** [comment]
Closing this for now as SDL requires no external repos prior to release. Moving the code to Gitlab and will fork and open a new PR when SDL has been fully completed.

**Pull request closed without merging**